### PR TITLE
feat!: add expedited propsals support

### PR DESCRIPTION
**Pull request by MSalopek**
Add expedited proposals to gaia v18.

Related:
* https://github.com/cosmos/cosmos-sdk/pull/20657
* https://github.com/cosmos/cosmos-sdk/pull/14720

Uses:
* https://github.com/informalsystems/wasmd/releases/tag/v0.45.0-lsm

### CODE REVIEW

**Comment by MSalopek** [comment]
Also requires changes to ICS to support the new changes in gov.

**Comment by MSalopek** [comment]
Also requires forking wasmd and supporting the correct gov interface.

File: go.mod [hunk 1]
```diff
@@ -231,14 +231,14 @@ replace (
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.2.0
 
 	// Use special SDK v0.47.x release with support for both ICS and LSM
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.47.15-ics-lsm
+	github.com/cosmos/cosmos-sdk => github.com/informalsystems/cosmos-sdk v0.46.0-alpha2.0.20240614081903-f1b9aac8d413
```
Comment (mpoke) [inline go.mod]: Why are we using a spacial 0.46 branch of SDK?

File: .changelog/unreleased/api-breaking/3149-bump-ics.md
Comment (mpoke) [inline .changelog/unreleased/api-breaking/3149-bump-ics.md:1]: Why is this under API breaking?

File: app/upgrades/v18/upgrades.go [hunk 2]
```diff
@@ -26,6 +29,16 @@ func CreateUpgradeHandler(
 			return vm, err
 		}
 
+		expeditedPeriod := 24 * 7 * time.Hour // 7 days
+		govParams := keepers.GovKeeper.GetParams(ctx)
+		govParams.ExpeditedVotingPeriod = &expeditedPeriod
+		govParams.ExpeditedThreshold = govv1.DefaultExpeditedThreshold.String() // 66.7%
+		govParams.ExpeditedMinDeposit = govParams.MinDeposit                    // full deposit amount is required
```
Comment (mpoke) [inline app/upgrades/v18/upgrades.go]: This needs to be 500 ATOMs. The min deposit is 250 ATOMs. See https://www.mintscan.io/cosmos/proposals/926

**Review by mpoke — COMMENTED** [review]
The general logic LGTM. The params are not initialized with the values from the proposal.

File: go.mod (hunk 1 above)
Comment (MSalopek) [reply go.mod]: Removed

File: .changelog/unreleased/api-breaking/3149-bump-ics.md
Comment (MSalopek) [reply .changelog/unreleased/api-breaking/3149-bump-ics.md:1]: Adds a new API method.

File: app/upgrades/v18/upgrades.go (hunk 2 above)
Comment (MSalopek) [reply app/upgrades/v18/upgrades.go]: Resolved in https://github.com/cosmos/gaia/pull/3149/commits/364db8496f98a7a31855730b3c0889834afe75e3

**Review by mpoke — APPROVED** [review]

**Review by dusan-maksimovic — APPROVED** [review]

**Pull request merged**